src/middle/Location.ml

```diff
@@ -75,6 +59,24 @@ let rec to_string ?printed_filename ?(print_file = true) ?(print_line = true)
     | None -> "" in
   sprintf "%s%scolumn %d%s" file line loc.col_num incl
 
+let compare loc1 loc2 =
+  let rec unfold = function
+    | {included_from= None; _} as loc -> [loc]
+    | {included_from= Some loc1; _} as loc2 ->
+        (* force tiebreaker to go to the file included from *)
+        loc2 :: unfold {loc1 with line_num= loc1.line_num + 1} in
```

Comment:
This is required due to the fact that comments/includes never call Lexing.token directly, so their position information is sometimes hairy. In particular, if two includes happen in a row the positions can all end up higher than they should from an included file.

The best example I can think of comes from running test/integration/rstanarm/binomial.stan without this line:
https://gist.github.com/WardBrian/d73f77b54339b3af5b621202b041a49c#file-binomial-bad-stan-L342

The functions block here has two includes in a row. Inserting anything in between them, even another comment, seems to fix the issue

It may be possible to avoid doing this if we really sort out 100% the issues with the preprocessor/lexer. The issue in the above is that the immediate next thing after the `#include` is another `#include`, so we never 100% reset the token positions. There's probably a similar hack available as what this currently does in finish_comment, but it's difficult to avoid 'artificially' inserting line breaks. If you're able to spend some time debugging I'd appreciate the help here